include/mqt-core/dd/Package.hpp

```diff
@@ -2203,6 +2223,320 @@ template <class Config> class Package {
+      // set fixed width to maximum of a printed number
+      // (-) 0.precision plus/minus 0.precision i
+      constexpr auto width = 1 + 2 + precision + 1 + 2 + precision + 1;
+      std::cout << ": " << std::setw(width)
+                << ComplexValue::toString(amplitude.r, amplitude.i, false,
+                                          precision)
+                << "\n";
+    }
+    std::cout << std::flush;
+  }
+
+  void printMatrix(const mEdge& e) {
+    const std::size_t element = 2ULL << e.p->v;
+    for (auto i = 0ULL; i < element; i++) {
+      for (auto j = 0ULL; j < element; j++) {
+        const auto amplitude = getValueByIndex(e, i, j);
+        constexpr auto precision = 3;
+        // set fixed width to maximum of a printed number
+        // (-) 0.precision plus/minus 0.precision i
+        constexpr auto width = 1 + 2 + precision + 1 + 2 + precision + 1;
+        std::cout << std::setw(width)
+                  << ComplexValue::toString(amplitude.r, amplitude.i, false,
+                                            precision)
+                  << " ";
+      }
+      std::cout << "\n";
+    }
+    std::cout << std::flush;
+  }
+
+  CMat getMatrix(const mEdge& e, std::size_t nrQubits) {
+    std::size_t dim = 0;
+    if (nrQubits != 0) {
+      dim = 2ULL << (nrQubits - 1);
+    }
+    // allocate resulting matrix
+    auto mat = CMat(dim, CVec(dim, {0.0, 0.0}));
+
+    // Identity case
+    if (e.isTerminal()) {
+      for (auto i = 0ULL; i < dim; i++) {
+        for (auto j = 0ULL; j < dim; j++) {
+          if (i == j) {
+            mat[i][j] = {1., 0.};
+          }
+        }
+      }
+    } else {
+      getMatrix(e, Complex::one, 0, 0, mat, static_cast<int>(nrQubits) - 1);
+    }
+    return mat;
+  }
+  void getMatrix(const mEdge& e, const Complex& amp, const std::size_t i,
+                 const std::size_t j, CMat& mat, const int level) {
+    // calculate new accumulated amplitude
+    auto c = e.w * amp;
+
+    std::size_t x = i;
+    std::size_t y = j;
+
+    if (level != -1) {
+      x = i | (1ULL << level);
+      y = j | (1ULL << level);
+    }
+
+    if (e.isTerminal() && level == -1) {
+      // base case
+      mat.at(i).at(j) = {c.r, c.i};
+      return;
+    }
+
+    if ((!e.isTerminal() && e.p->v == level)) {
+      // recursive case
+      if (!e.p->e[0].w.approximatelyZero()) {
+        getMatrix(e.p->e[0], c, i, j, mat, level - 1);
+      }
+      if (!e.p->e[1].w.approximatelyZero()) {
+        getMatrix(e.p->e[1], c, i, y, mat, level - 1);
+      }
+      if (!e.p->e[2].w.approximatelyZero()) {
+        getMatrix(e.p->e[2], c, x, j, mat, level - 1);
+      }
+      if (!e.p->e[3].w.approximatelyZero()) {
+        getMatrix(e.p->e[3], c, x, y, mat, level - 1);
+      }
+    } else if ((!e.isTerminal() && e.p->v < level) ||
+               (e.isTerminal() && level != -1)) {
+      getMatrix(e, c, i, j, mat, level - 1);
+      getMatrix(e, c, x, y, mat, level - 1);
+    }
+  }
+
```

Comment:
```suggestion
```
These can be removed completely, as they have all been moved to the `Edge.hpp` file. 
Any adaptation should happen there.